test_unstructured/partition/test_html_partition.py

```diff
@@ -723,3 +726,43 @@ def test_partition_html_with_table_without_tbody(tag: str, expected: str):
     )
     partitions = partition_html(text=table_html)
     assert partitions[0].metadata.text_as_html == expected
+
+
+@pytest.fixture
+def partitioned_html_with_duplicate_elements():
+    filename = "example-docs/fake-html-duplicates.html"
+    with open(filename) as f:
+        elements = partition_html(file=f)
+    return elements
+
+
+def test_each_element_has_page_number_and_index_metadata(partitioned_html_with_duplicate_elements):
+    for element in partitioned_html_with_duplicate_elements:
+        assert element.metadata.page_number is not None
+        assert element.metadata.index_on_page is not None
+
+
+def test_all_element_ids_are_unique(partitioned_html_with_duplicate_elements):
+    assert convert_to_dataframe(partitioned_html_with_duplicate_elements).element_id.is_unique
```

Comment:
seems the long way around the barn to use `convert_to_dataframe()`, also that couples the test to a dependency outside the unit under test. Could be:
```python
ids = [e.id for e in partition_html("example-docs/fake-html-duplicates.html")]
assert len(ids) == len(set(ids))
```

Reply:
done, no longer using `conver_to_dataframe`